core/src/main/java/org/apache/accumulo/core/iterators/user/ScanYieldException.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.iterators.user;
+
+import org.apache.accumulo.core.data.Key;
+
+/**
+ * This exception can be thrown at from a next or seek call on an iterator to allow other scans to get time slots. This mechanism is intended to avoid a set of
+ * scans to dominate all of the scan slots (readahead threads) and starve other scans out.
+ */
+public class ScanYieldException extends RuntimeException {
+  private final Key position;
+
+  public ScanYieldException(Key position) {
```

Comment:
Can we make this more specific as to what `Key` should be passed in? If I understand correctly:

1) For seek(), the start key of the Range passed to the Iterator
2) For next(), the last key returned by the Iterator

Maybe there is a third case for when seek() also consumes data before returning the call (finding the first key in the range which matches some condition)? I'm not sure of a concise way to state that :)